docs/internals/optimizer.rst

```diff
@@ -1121,6 +1122,51 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
```

Comment:
```suggestion
.. index:: ! unused store eliminator
.. _unused-store-eliminator:
```

Reply:
Why though, as I've already put it up top in the `index` section?

Reply:
Yeah, but `!` means it's the main entry for the topic.

Also, putting it at the top is not great IMO and I'd remove it. When you go to the index, you want to be taken to the section that's about your keyword, not the beginning of a big page that has that somewhere :)

Reply:
I'd prefer to do this in a separate PR (we can even open an issue for this), since none of anchors (or indices) in `optimizer.rst` use this approach.